### DISCUSSION TITLE
Virtualized slides midway freeze during swipe

### DISCUSSION THREAD

**Discussion by ajaypillay** [nolimits4web/swiper#4661] [Q&A]
Hi, so I have some rather complex components being rendered in a virtualized swiper in React.

The issue is that as I swipe, there is a momentary freeze, because the virtualized swiper is updating a new slide to render `onSlideChange`.

Is there any way to **force** the virtualized swiper to render a new slide `onSlideTransitionEnd` instead? Because then due to the state update being done **after** the animation completes, there won't be a freeze.

It's a little hard to reproduce the code in a sandbox due to the complexity of the components.

EDIT: I have recordings of it happening here:

Without virtualization (only 5 slides here so it's smooth, it actually contains more but it's too slow and choppy if I render more without virtualization):
https://user-images.githubusercontent.com/917752/121326767-6e933680-c945-11eb-898d-2bc7390100cd.mp4

With virtualization:

https://user-images.githubusercontent.com/917752/121326836-7e127f80-c945-11eb-9143-7325a201b2aa.mp4

You can see that there is a momentary freeze whenever I swipe to a new slide. I have tried increasing the rendered slide window by increasing `addSlidesAfter` and `addSlidesBefore` but that just made it even more choppy.

My best guess is that the virtualization is causing the slide to be rendered `onSlideChange` instead of `onSlideTransitionEnd`, that's why there's the momentary freeze.

Is there any way for me to change/fix this?

**Comment by 3zzy** [comment]
Were you able to resolve this? Would be great if you could post your solution here for everyone else. Thanks!

**Reply by ajaypillay** [reply]
No, it's still open as I've not been able to resolve this specifically. However, I don't have freezing anymore but I cannot pinpoint exactly what change fixed this.

I did some rework with regards to my API fetching the data to populate this virtual list. And right now I fetch more data upon `onSlideChangeTransitionEnd` and this way freezes seem to have gone away.

**Comment by dev-aly3n** [comment]
any update?

**Comment by dev-aly3n** [comment]
> Is there any way to force the virtualized swiper to render a new slide onSlideTransitionEnd instead? Because then due to the state update being done after the animation completes, there won't be a freeze.

you're right. it will fix the issue
